src/main/resources/ath-container/Dockerfile

```diff
@@ -37,6 +37,13 @@ RUN install -m 0755 -d /etc/apt/keyrings \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
 
+# Despite the docker SUID hack below, test-containers accesses /var/run/docker.sock directly and so we can not rely on the SUID hack.
+# Rather take the docker user group as an arg and make the ath-user a member of that group
+# we retain the suid workaround as this method requires a local build of the container
+# we need to do this before we install docker so that any files have the correct permission
+ARG dockergid=1002
```

Comment:
I think the default group ID should match the one used in practice by our production setup (131, I think). Using a default group ID that doesn't match the Ubuntu default seems like it will be a maintenance burden.

Reply:
The default in my Ubuntu setup is 1001...  
There is no default as it depends on the order that packages may be installed.

Reply:
Interesting. On my Ubuntu 22.04 desktop it's 131. In a fresh `noble` Docker image (installing the `docker.io` package) it's 103. I think you are right that there is no default.

Reply:
OH: interesting. The `999` user is created by one of our early `Dockerfile` steps:

```shell
$ docker run --rm -ti --entrypoint=grep --platform=linux/amd64 ubuntu:noble 999 /etc/group
$ docker run --rm -ti --entrypoint=grep --platform=linux/amd64 jenkins/ath 999 /etc/group
systemd-journal:x:999:
```

That means removing it should not be a problem

Reply:
Confirmed: the user `systemd-journal` is created when installing the `openjdk-*-jdk` packages. I believe they have a dependency to a  `systemd-journal*` package.

- I'm adding a suggestion with a short term fix (deleting the group if existing as we don't really care about this one)
- WDYT about using the temurin JDKs instead of the `openjdk` packages in this context?
  - Faster image build as we could use multi stage builds (parallel)
  - Closer to the rest of the environment

Reply:
See https://github.com/jenkinsci/acceptance-test-harness/pull/1701/files#r1744863012

Reply:
removing system-d (or other groups) does not fill me with confidence.

but all groups created by apt should be dynimcally allocated an id (should not be hard coded) so if I move the group creation first it should be able to do the right thing in all cases

Reply:
SystemD is not expected to be run inside a container unless you do terrible hacks. 
since we tracked this group as a consequence of installing the ubuntu openjdk packages, which can be replaced by temurin standalone jdks, you can do it safely.

however you are correct : if you create the group before the apt installs, it should be good 👍🏼

Reply:
https://github.com/jenkinsci/acceptance-test-harness/pull/1705/commits/f336732785122935d85a2b5f847c6cf699138c0a (different PR as the Jenkinsfile change needs to be from a trusted fork)